File: src/mqtt/topics.js
/**
 *宋建国：2019/11/28 16:48
 *描述：数据交换协议
 *版本：1.0
 */
//IM消息
export const $IMNEWS = "$XIM/NEWS/";
  //系统广播消息
export const $IMBROADCAST = "$XIM/BROADCAST";
  //推送页面消息
export const $PAGENEWS = "$PAG/NEWS/";
  //系统广播页面消息
export const   $PAGEBROADCAST = "$PAG/BROADCAST";
  //消息服务器系统消息
export const $SYS = "$SYS/";
//与微服务直接通信主题
//格式：$WFW/服务名称
export const $WFW = "$WFW/"
//消息发送类型
export const $MESSAGETYPE = {
  ChatText: "0", //文本
  ChatImage: "1",//图片
  ChatFile: "2", //文件
  ChatVoice: "3", //语音
  ChatVideo: "4", //小视频
  Addfriends: "20", //好友确认消息
  ApplyFriend: "21", //申请加好友
  CreateGroup: "22", //创建群消息
  QuitGroup: "23",//退出群消息
  InGroup: "24", //进群消息
}

export default {
  $IMNEWS,
  $IMBROADCAST,
  $PAGENEWS,
  $PAGEBROADCAST,
  $SYS,
  $WFW,
  $MESSAGETYPE
}

File: src/main/java/com/xyjsoft/core/mqtt/topics.java
package com.xyjsoft.core.mqtt;

/**
 * @author sjg
 * 2019年12月3日 上午10:30:18  
 * MQTT 交互协议
 */
public class topics {
	/**
	 * @describe IM消息
	 * @author sjg 
	 * @date 2019年12月3日 上午10:31:02
	 */
	public static String $IMNEWS = "$XIM/NEWS/";
	/**
	 * @describe 系统广播消息
	 * @author sjg 
	 * @date 2019年12月3日 上午10:31:12
	 */
	public static String $IMBROADCAST = "$XIM/BROADCAST";
	/**
	 * @describe 推送页面消息
	 * @author sjg 
	 * @date 2019年12月3日 上午10:31:24
	 */
	public static String $PAGENEWS = "$PAG/NEWS/";
	/**
	 * @describe 系统广播页面消息
	 * @author sjg 
	 * @date 2019年12月3日 上午10:31:36
	 */
	public static String $PAGEBROADCAST = "$PAG/BROADCAST";
	/**
	 * @describe 消息服务器系统消息
	 * @author sjg 
	 * @date 2019年12月3日 上午10:31:47
	 */
	public static String $SYS = "$SYS/";
	
	/**
	 * @describe 微服务通信协议
	 * 每个微服务唯一通信主题格式：$WFW/服务名称
	 * @author sjg 
	 * @date 2019年12月3日 上午10:38:43
	 */
	public static String $WFW = "$WFW/";
}
